semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -752,7 +773,9 @@ and m_expr a b =
       (match H.name_of_dot_access a with
       | None -> fail ()
       | Some a1 -> m_name a1 b1)
-      >||> m_with_symbolic_propagation (fun b1 -> m_expr a b1) b
```

Comment:
May be a bit cleaner to pass `is_root` to `m_with_symbolic_propagation` and then use it in the condition of the `if_config` there.